src/auth_service/auth_adapter/adapters/memory_session_store.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""An adapter for a memory based session store."""
+
+import asyncio
+from contextlib import suppress
+from typing import Optional
+
+from auth_service.auth_adapter.core.session_store import (
+    Session,
+    SessionConfig,
+    SessionStore,
+)
+
+
+class MemorySessionStore(SessionStore):
+    """Memory based store for user sessions."""
+
+    def __init__(
+        self,
+        *,
+        config: SessionConfig,
+    ):
+        """Initialize the memory based session store."""
+        super().__init__(config=config)
+        self.lock = asyncio.Lock()
+        self.store: dict[str, Session] = {}
+
+    async def create_session(self) -> Session:
+        """Create a new user session in the store and return it."""
+        for _ in range(100):
+            async with self.lock:
+                session = self._create_session()
+                # avoid overwrite an existing session
+                if session.session_id not in self.store:
+                    self.store[session.session_id] = session
+                    return session
+            await asyncio.sleep(0)
```

Comment:
What's the reason to have that in here?

Reply:
Just because it's a tight loop.

Actually, thinking about it, we do not even need a lock since no task switch can happen in that block. Will change it.